### PR TITLE
Port dynamic object visualization

### PR DESCRIPTION
**Pull request by esteve**

### CODE REVIEW

**Comment by esteve** [comment]
@jilaada I recall you ported code that used `unique_id`, I had to use `unique_identifier_msgs` instead and copied code from `unique_id`, not sure if that's the correct way. Could you have a look at this PR to make sure the changes are ok? Thanks.

**Review by jilaada — COMMENTED** [review]
Some minor comments. I was able to compile (after using rosdep to install any missing dependencies) and run the node via the launch file. While I though the `anon` keyword would not be supported it seems that running two nodes will output an automatically generated node name: 

```
jilada@ade:~$ ros2 node list
/dynamic_object_visualization_ade_3801_3408396284551468616
/dynamic_object_visualization_ade_3905_4584043922755780744
```

File: perception/util/visualizer/dynamic_object_visualization/CMakeLists.txt [hunk 1]
```diff
@@ -1,46 +1,22 @@
...
-    visualization_msgs
-)
-
-include_directories(include ${catkin_INCLUDE_DIRS})
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Wno-unused-parameter -Werror)
```
Comment (jilaada) [inline perception/util/visualizer/dynamic_object_visualization/CMakeLists.txt:12]: I think we all agreed that `-Werror` would not be included in this compile flags due to how strict it is

File: perception/util/visualizer/dynamic_object_visualization/CMakeLists.txt [hunk 2]
```diff
@@ -1,46 +1,22 @@
...
-  CATKIN_DEPENDS
-    roscpp
-    autoware_perception_msgs
-    geometry_msgs
-    sensor_msgs
-    visualization_msgs
-)
-
-include_directories(include ${catkin_INCLUDE_DIRS})
+### Compile options
```
[the remark this answers is not in the document]
Comment (esteve) [reply perception/util/visualizer/dynamic_object_visualization/CMakeLists.txt:4]: Done in 454825d

File: perception/util/visualizer/dynamic_object_visualization/src/dynamic_object_visualizer.cpp [hunk 3]
```diff
@@ -48,38 +53,38 @@ void DynamicObjectVisualizer::dynamicObjectWithFeatureCallback(
 }
 
 void DynamicObjectVisualizer::dynamicObjectCallback(
-  const autoware_perception_msgs::DynamicObjectArray::ConstPtr & input_msg)
+  const autoware_perception_msgs::msg::DynamicObjectArray::ConstSharedPtr input_msg)
 {
-  if (pub_.getNumSubscribers() < 1) return;
-  visualization_msgs::MarkerArray output;
+  if (this->count_subscribers("output") < 1) return;
```
[the remark this answers is not in the document]
Comment (esteve) [reply perception/util/visualizer/dynamic_object_visualization/src/dynamic_object_visualizer.cpp]: Done in 9339e16

File: perception/util/visualizer/dynamic_object_visualization/CMakeLists.txt (hunk 1 above)
Comment (esteve) [reply perception/util/visualizer/dynamic_object_visualization/CMakeLists.txt]: Removed in 9e3e4ad, though the code seems to pass with `-Werror` enabled.

File: perception/util/visualizer/dynamic_object_visualization/src/dynamic_object_visualizer.cpp [hunk 4]
```diff
@@ -136,19 +141,28 @@ void DynamicObjectVisualizer::dynamicObjectCallback(
...
     marker.ns = std::string("label");
     std::string label;
     if (!getLabel(input_msg->objects.at(i).semantic, label)) continue;
     marker.scale.x = 0.5;
     marker.scale.z = 0.5;
-    std::string id_str = unique_id::toHexString(input_msg->objects.at(i).id);
+
+    // NOTE(esteve): copied from
+    // https://github.com/ros-geographic-info/unique_identifier/blob/master/unique_id/include/unique_id/unique_id.h#L177-L182
+    boost::uuids::uuid uu;
```
[the remark this answers is not in the document]
Comment (esteve) [reply perception/util/visualizer/dynamic_object_visualization/src/dynamic_object_visualizer.cpp]: Thanks for the link. I've modified the code to make it simpler in 41512076d62ceeebea3e039230e33fcee7db8c60 and removed the dependency on Boost.

File: perception/util/visualizer/dynamic_object_visualization/src/dynamic_object_visualizer.cpp [hunk 5]
```diff
@@ -555,19 +570,27 @@ bool DynamicObjectVisualizer::getLabel(
 }
 
 void DynamicObjectVisualizer::getColor(
-  const autoware_perception_msgs::DynamicObject & object, std_msgs::ColorRGBA & color)
+  const autoware_perception_msgs::msg::DynamicObject & object, std_msgs::msg::ColorRGBA & color)
 {
-  std::string id_str = unique_id::toHexString(object.id);
+  // NOTE(esteve): copied from
+  // https://github.com/ros-geographic-info/unique_identifier/blob/master/unique_id/include/unique_id/unique_id.h#L177-L182
+  boost::uuids::uuid uu;
```
[the remark this answers is not in the document]
Comment (esteve) [reply perception/util/visualizer/dynamic_object_visualization/src/dynamic_object_visualizer.cpp]: Done in https://github.com/tier4/Pilot.Auto/commit/41512076d62ceeebea3e039230e33fcee7db8c60

**Review by jilaada — APPROVED** [review]
Changes look good to me. Thanks for the making the requested changes

**Pull request merged**